.github/workflows/commands.yml

```diff
@@ -1,131 +1,182 @@
-        git push origin HEAD:"refs/heads/$PR_HEAD_REF"\
-    - name: Post command comment
-      if: steps.fantomas.outcome == 'success' || steps.xlf.outcome == 'success' || steps.ilverify.outcome == 'success'
-      uses: actions/github-script@v3
-      with:
-        script: |
-          // Probably, there's more universal way of getting outputs, but my gh-actions-fu is not that good.
-          var output = ""
-          if ("${{steps.command-extractor.outputs.result}}" == 'fantomas') {
-            output = "${{steps.fantomas.outputs.result}}"
-          } else if ("${{steps.command-extractor.outputs.result}}" == 'xlf') {
-            output = "${{steps.xlf.outputs.result}}"
-          } else if ("${{steps.command-extractor.outputs.result}}" == 'ilverify') {
-            output = "${{steps.ilverify.outputs.result}}"
-          }
-          const body = `Ran ${{ steps.command-extractor.outputs.result }}: https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${process.env.GITHUB_RUN_ID}\n${output}`;
-          await github.issues.createComment({
-            issue_number: context.issue.number,
-            owner: context.repo.owner,
-            repo: context.repo.repo,
-            body: body
-          });
-    - name: Post command failed comment
-      if: failure()
-      uses: actions/github-script@v3
-      with:
-        script: |
-          const body = `Failed to run ${{ steps.command-extractor.outputs.result }}: https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${process.env.GITHUB_RUN_ID}`;
-          await github.issues.createComment({
-            issue_number: context.issue.number,
-            owner: context.repo.owner,
-            repo: context.repo.repo,
-            body: body
-          });
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+          
+      - name: Checkout PR branch
+        run: gh auth setup-git && gh pr checkout ${{ github.event.issue.number }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: cli-results
+
+      - name: Read metadata
+        id: read-meta
+        run: |
+          source result
+          echo "run_step_outcome=$run_step_outcome" >> $GITHUB_OUTPUT
+          echo "hasPatch=$hasPatch" >> $GITHUB_OUTPUT
+
+      - name: Apply and push patch
+        if: ${{ steps.read-meta.outputs.run_step_outcome == 'success' && steps.read-meta.outputs.hasPatch == 'true' }}
+        run: |
+          patch -p1 -s --force < repo.patch || true
+          git config user.name "GH Actions"
+          git config user.email "actions@github.com"          
+          git add -u
+          git commit -m "Apply patch from ${{ needs.detect-and-run.outputs.command }}"
```

Comment:
Important: The `command` must be from a known set, otherwise this is shell injection.
The command is coming from the comment pipeline, only valid command will go trough.